R/lazyframe__lazy.R

```diff
@@ -270,6 +270,27 @@ LazyFrame_collect_background = function() {
   .pr$LazyFrame$collect_background(self)
 }
 
+#' @title Collect_in_background
+#' @description collect DataFrame by lazy query in a background thread.
+#' @details
+#' Useful to not block the R session while query executes. Any use of `$map()` or `$apply()` is
+#' diverted to other R sessions. If R functions rely on global variables this can change the
```

Comment:
I just added this to docs "Any use of $map() and $apply() will be diverted to other R session" That was not entirely correct. Error messages when trying to use map and apply with `collect_in_background` are very poor.

it seems map(in_background=TRUE) does not work with `collect_inbackground`

```R
pl$set_global_rpool_cap(1)

lf = pl$LazyFrame(iris)

#no informative error, improve
h = lf$groupby("Species")$agg(
  pl$col("Sepal.Length")$apply(\(s) s$to_r() |> length())
)$collect_in_background()

#poor error message, improve.
h$join()

#this works
h = lf$groupby("Species")$agg(
  pl$col("Sepal.Length")$apply_in_background(\(s) s$to_r() |> length())
)$collect_in_background()
h$join()

#this does not work, why not ?
h = pl$LazyFrame(iris)$select(pl$col("Species")$map(\(x) length(x)),in_background =TRUE)$collect_in_background()

```

Reply:
Maybe we should remove `collect_inbackgrounnd` to avoid confusion

Reply:
I don't think this has been resolved. Using collect_in_background gives to me surprising behavior and not clear error messages (see above)

Reply:
The final statement is broken because you have wrong parenthesis. The `in_background` argument is not passed to `map`.

Reply:
I've improved the error handling in background:
``` r
> th <- pl$LazyFrame(mtcars)$select(pl$col("mpg")$map(\(x) y, in_background = TRUE))$collect_in_background()
> th$join()
Error: Execution halted with the following contexts
   0: During function call [th$join()]
   1: When trying to map a polars series with R function in the background R process
   2: EvalError(lang!(function (x) y, ExternalPtr.set_class(["Series"]))
```